go/vt/vtgate/planbuilder/operators/joins.go

```diff
@@ -37,8 +37,6 @@ func AddPredicate(
 	ctx *plancontext.PlanningContext,
 	join JoinOp,
 	expr sqlparser.Expr,
-	joinPredicates bool,
-	newFilter func(Operator, sqlparser.Expr) Operator,
```

Comment:
There was only one place where we were calling with JoinPredicates value set to true, and that I made a separate function as stated in ☝️. So, I could refactor this as well.